pallets/gear/src/lib.rs

```diff
@@ -962,11 +962,38 @@ pub mod pallet {
                 .unwrap_or(false)
         }
 
-        /// Returns true if a program has terminated status
+        /// Returns true if id is a program and the program has active status.
+        pub fn is_active(program_id: ProgramId) -> bool {
+            common::get_program(program_id.into_origin())
+                .map(|p| p.is_active())
+                .unwrap_or_default()
+        }
+
+        /// Returns true if id is a program and the program has terminated status.
         pub fn is_terminated(program_id: ProgramId) -> bool {
             common::get_program(program_id.into_origin())
                 .map(|p| p.is_terminated())
-                .unwrap_or(false)
+                .unwrap_or_default()
+        }
+
+        /// Returns true if id is a program and the program has exited status.
+        pub fn is_exited(program_id: ProgramId) -> bool {
+            common::get_program(program_id.into_origin())
+                .map(|p| p.is_exited())
+                .unwrap_or_default()
+        }
+
+        /// Returns true if id is a program and the program has exited status.
+        pub fn exit_argument_of(program_id: ProgramId) -> Option<ProgramId> {
+            common::get_program(program_id.into_origin())
+                .map(|p| {
```

Comment:
seems it's case for `and_then` instead of calling `unwrap_or_default` with `map`?

Reply:
If I'm not mistaken, `and_then` returns Option anyway, so we have to unwrap it further

Reply:
I guess it doesn't make nested Options.

Reply:
> I guess it doesn't make nested Options.

Yeah, it doesn't, but for me that's the same code:
```rust
// #1
get_program(program_id.into_origin())
    .map(|p| p.is_active())
    .unwrap_or_default()
// #2
get_program(program_id.into_origin())
    .and_then(|p| p.is_active())
    .unwrap_or_default()
```

Reply:
oops, got confused of `and_then`. I thought much of it, the it really was